create_per_gene_matrix.py

```diff
@@ -253,69 +308,61 @@ def compute_from_full_mt(chr20: bool, overwrite: bool):
         'gene_id'
     ).aggregate_rows(
         gene_symbol=hl.agg.take(mt.gene_symbol, 1)[0]
-    ).aggregate(
-        counts=hl.agg.filter(
-            hl.if_else(
-                mt.pop == 'all',
-                hl.is_defined(mt.popmax) & (mt.popmax.AF <= MAX_FREQ),
-                mt.freq[freq_dict[mt.pop]].AF <= MAX_FREQ
-            ),
-            hl.agg.group_by(
-                hl.if_else(
-                    mt.pop == 'all',
-                    mt.popmax.AF > 0.001,
-                    mt.freq[freq_dict[mt.pop]].AF > 0.001
-                ),
-                hl.struct(
-                    hom_csq=hl.agg.filter(~mt.is_het, hl.agg.min(mt.csq)),
-                    het_csq=hl.agg.filter(mt.is_het, hl.agg.min(mt.csq)),
-                    het_het_csq=hl.sorted(
-                        hl.array(
-                            hl.agg.filter(mt.is_het, hl.agg.counter(mt.csq))
-                        ),
-                        key=lambda x: x[0]
-                    ).scan(
-                        lambda i, j: (j[0], i[1] + j[1]),
-                        (0, 0)
-                    ).find(
-                        lambda x: x[1] > 1
-                    )[0]
+	).aggregate(
+    **{f"af_le_{af}": hl.agg.filter(
+                    hl.is_defined(mt.popmax) & 
+                    (mt.popmax <= MAX_FREQ),
+                hl.agg.group_by(
+                        mt.popmax <= af,
+                    hl.struct(
+                        hom_csq=hl.agg.filter(~mt.is_het, hl.agg.collect_as_set(mt.cum_csq)),
+                        het_csq=hl.agg.filter(mt.is_het, hl.agg.collect_as_set(mt.cum_csq))
+                    )
                 )
             )
-        )
+        for af in ALLELE_FREQUENCY_CUTOFFS
+    }
     )
-
+    
     mt = mt.annotate_entries(
-        counts=hl.struct(
-            all=hl.struct(
-                hom_csq=hl.min(mt.counts.get(True).hom_csq, mt.counts.get(False).hom_csq),
-                het_csq=hl.min(mt.counts.get(True).het_csq, mt.counts.get(False).het_csq),
-                het_het_csq=hl.min(
-                    mt.counts.get(True).het_het_csq,
-                    mt.counts.get(False).het_het_csq,
-                    hl.or_missing(
-                        hl.is_defined(mt.counts.get(True).het_csq) & hl.is_defined(mt.counts.get(False).het_csq),
-                        hl.max(mt.counts.get(True).het_csq, mt.counts.get(False).het_csq)
-                    )
-                ),
-            ),
-            af_le_0_001=mt.counts.get(False)
-        )
+            **{f"af_le_{af}":hl.struct(
+                hom_csq=mt[f"af_le_{af}"].get(True).hom_csq,
+                het_csq=mt[f"af_le_{af}"].get(True).het_csq
+            )
+            for af in ALLELE_FREQUENCY_CUTOFFS
+            }
     )
 
-    mt = mt.checkpoint('gs://gnomad-tmp/compound_hets/het_and_hom_per_gene{}.1.mt'.format(
+    mt = mt.checkpoint('gs://gnomad-tmp/compound_hets/het_and_hom_per_gene{}.2.mt'.format(
```

Comment:
Might want to repartition here, you can write the checkpoint and then repartition on read with `_n_partitions`